docs/LANGUAGE.md

```diff
@@ -0,0 +1,388 @@
+# Opalog: OPA's Query Language
+
+OPA includes a policy engine which is purpose built for reasoning about information represented in structured documents such as JSON. Data stored in in the policy engine can be queried using OPA's native query language: Opalog.
+
+## What is Opalog?
+
+Opalog is a modernized version of [Datalog](https://en.wikipedia.org/wiki/Datalog) which is a well understood, decades old query language. Opalog extends Datalog to support structured document models such as JSON. Opalog queries are assertions on data stored in OPA. These queries can be used to define policies which enumerate violations that exist in the data stored in OPA.
+
+## Why use Opalog?
+
+Use Opalog for defining policy which is easy to read and write. 
+
+Opalog focuses on providing powerful support for referencing nested documents and ensuring that queries are correct and unambiguous.
+
+Opalog is declarative so policy authors can focus on what queries should return rather than how queries should be executed. These queries are simpler and more concise than the equivalent in an imperative language. Like other applications which support declarative query languages, OPA is able to optimize queries to improve performance, e.g, indexing, concurrent evaluation, reordering, etc.
+
+## Opalog Grammar
+
+Opalog's syntax is defined by the following grammar:
+
+```
+module         ::= package { import } policy
+package        ::= "package" var { "." var }
+import         ::= "import" package [ "as" var ]
+policy         ::= { rule }
+rule           ::= rule-head ":-" rule-body
+rule-head      ::= rule-name [ rule-args ]
+rule-name      ::= var
+rule-args      ::= "[" [ key ] "]" [ "=" value ]
+key            ::= var
```

Comment:
Notationally, is there another symbol we can use besides [] to mean "optional"?  Isn't a question-mark standard?  And isn't an asterisk standard for repetition?  I find the use of [] and {} confusing since I'm accustomed to ?*, and since [] and {} are meaningful symbols in our language.